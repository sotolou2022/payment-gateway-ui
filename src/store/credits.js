import axios from "axios";
import { defineStore } from "pinia";

const baseUrl = "http://localhost:5000/api/payment/";
const creditUrl = "api/payment/getAllCredits";

export const useCreditStore = defineStore({
  id: "credit",
  state: () => ({
    credits: [],
    token: "",
    errorMessage: null,
  }),
  getters: {
    getCredits: (state) => {
      return state.credits;
    },
  },
  actions: {
    async fetchCredits() {
      console.log("fetchCredits!! ");
      this.credits = [];

      try {
        const creditResponse = await axios.get(creditUrl);

        if (creditResponse) {
          this.credits = creditResponse.data;
          console.log("credit data: " + this.credits);
        }
      } catch (error) {
        this.errorMessage = error;

        if (error.response.status === 401) {
          console.log("error message: " + this.errorMessage);
          // re-auth if 401 error
        }

        return Promise.reject(error.response.data);
      }
    },
  },
});
