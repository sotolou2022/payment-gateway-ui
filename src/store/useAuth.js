import { defineStore } from "pinia";
import axios from "axios";

const baseUrl = "http://localhost:5000/api/";

export const useAuthStore = defineStore("auth", {
  state: () => ({
    loggedIn: localStorage.getItem("token") ? true : false,
    // user: null,
    users: null,
    username: "",
    user: {},
  }),

  getters: {},

  actions: {
    async login(credentials) {
      console.log(" login:: " + JSON.stringify(credentials));
      this.username = credentials.username;
      // call authenticate
      const response = (await axios.post("api/authenticate", credentials)).data;
      console.log(" response:: " + JSON.stringify(response));

      if (response) {
        const token = `Bearer ${response.id_token}`;

        localStorage.setItem("token", token);
        axios.defaults.headers.common["Authorization"] = token;

        await this.ftechUser();
      }
    },

    async logout() {
      // const response = (await axios.post("api/logout")).data;

      // if (response) {
      localStorage.removeItem("token");

      this.$reset();
      // }
    },

    async ftechUser() {
      console.log("fetchUser!! ");
      const response = await axios.get("api/users/all");

      if (response.status === 200) {
        this.users = response.data;
        // this.users = (await axios.get("api/users/all")).data;
        if (this.users) {
          this.user = this.users.find(
            (user) => user.username === this.username
          );
          console.log("fetchUser: " + JSON.stringify(this.users));
          this.loggedIn = true;
        } else {
          console.log("Error: no users !! ");
          this.logout();
        }
      }
    },
  },
});
