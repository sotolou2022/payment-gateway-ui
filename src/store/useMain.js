import { defineStore } from "pinia";

export const useMainStore = defineStore({
  id: "main",
  state: () => ({
    hideConfigButton: false,
    isPinned: true,
    showConfig: false,
    isTransparent: "bg-transparent",
    isRTL: false,
    color: "",
    isNavFixed: false,
    isAbsolute: false,
    showNavs: true,
    showSidenav: true,
    showNavbar: true,
    showFooter: true,
    showMain: true,
    navbarFixed:
      "position-sticky blur shadow-blur left-auto top-1 z-index-sticky px-0 mx-4",
    absolute: "position-absolute px-4 mx-0 w-100 z-index-2",
    bootstrap,
  }),
  getters: {
    getCredits: (state) => {
      return state.credits;
    },
    navClasses: (state) => {
       return {
        "position-sticky blur shadow-blur mt-4 left-auto top-1 z-index-sticky": state.isNavFixed,
        "position-absolute px-4 mx-0 w-100 z-index-2": state.isAbsolute,
        "px-0 mx-4 mt-4": !state.isAbsolute,
      };
    }
  },
  actions: {
    toggleConfigurator() {
      this.showConfig = !this.showConfig;
    },
    navbarMinimize() {
      const sidenav_show = document.querySelector(".g-sidenav-show");
      if (sidenav_show.classList.contains("g-sidenav-hidden")) {
        sidenav_show.classList.remove("g-sidenav-hidden");
        sidenav_show.classList.add("g-sidenav-pinned");
        this.isPinned = true;
      } else {
        sidenav_show.classList.add("g-sidenav-hidden");
        sidenav_show.classList.remove("g-sidenav-pinned");
        this.isPinned = false;
      }
    },
    sidebarType(payload) {
      this.isTransparent = payload;
    },
    cardBackground(payload) {
      this.color = payload;
    },
    navbarFixed() {
      if (this.isNavFixed === false) {
        this.isNavFixed = true;
      } else {
        this.isNavFixed = false;
      }
    },
    toggleEveryDisplay() {
      this.showNavbar = !this.showNavbar;
      this.showSidenav = !this.showSidenav;
      this.showFooter = !this.showFooter;
    },
    toggleHideConfig() {
      this.hideConfigButton = !this.hideConfigButton;
    },
    toggleSidebarColor( payload) {
        // commit("sidebarType", payload);
        this.isTransparent = payload;
      },
      setCardBackground( payload) {
        // commit("cardBackground", payload);
        this.color = payload;
      },

  },
});
