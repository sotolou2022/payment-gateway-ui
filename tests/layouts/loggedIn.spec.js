import { mount, enableAutoUnmount } from "@vue/test-utils";
import { beforeEach, afterEach, expect } from "@jest/globals";
import Component from "~/layouts/loggedIn";
import { createTestingPinia } from "@pinia/testing";
import { useAuthStore } from "~/store/useAuth";
import authPlugin from "~/plugins/auth"

let wrapper;

beforeEach(() => {
  wrapper = mount(Component, { global: { plugins: [createTestingPinia(), authPlugin] } });
});

enableAutoUnmount(afterEach);

describe("Test suite for loggedIn layout", () => {
  it("is a vue component", () => expect(wrapper.vm).toBeTruthy());

  it("match snapshot when user is logged out", () => expect(wrapper.html()).toMatchSnapshot());

  it("match snapshot when user is logged in", () => {
    const auth = useAuthStore();

    auth.$state = {
      loggedIn: true,
      user: {
        id: 1,
        name: "Lindrit Krasniqi",
        email: "lindritkrasniqi@example.com",
      },
    };

    expect(wrapper.html()).toMatchSnapshot();
  });
});
