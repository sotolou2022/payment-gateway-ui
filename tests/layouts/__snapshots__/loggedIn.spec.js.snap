// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for loggedIn layout match snapshot when user is logged in 1`] = `"<div class=\\"container d-flex justify-content-center align-items-center\\" style=\\"height: 100vh;\\"></div>"`;

exports[`Test suite for loggedIn layout match snapshot when user is logged out 1`] = `"<div class=\\"container d-flex justify-content-center align-items-center\\" style=\\"height: 100vh;\\"></div>"`;
